core/kits/assets/js/commands/back.js

```diff
@@ -41,6 +53,72 @@ export class Back extends $e.modules.CommandBase {
 
 		return this.confirmDialog;
 	}
+
+	isGlobalRoute() {
+		const panelHistory = $e.routes.getHistory( 'panel' ),
+			isGlobal = /global\/\bglobal-colors|global-typography\b/.test( panelHistory[ panelHistory.length - 1 ].route );
+
+		if ( isGlobal ) {
+			const kit = elementor.config.kit_id;
+			this.document = elementor.documents.get( kit );
+		}
+
+		return isGlobal;
+	}
+
+	isDocumentChanged() {
+		return new Promise( ( resolve ) => {
+			if ( ! this.document || ! this.document.editor.isChanged ) {
+				resolve();
+				return;
+			}
+
+			this.getUnsavedChangesDialog( resolve ).show();
+		} );
```

Comment:
Why promise? where is the async function?

Reply:
No async function. The promise is because it can be resolved either from isDocumentChanged or the dialog.

Reply:
Can this be resolved?